message_ix/testing/__init__.py

```diff
@@ -476,7 +477,25 @@ def make_westeros(mp, emissions=False, solve=False, quiet=True):
         year=model_horizon,
     )
 
-    gdp_profile = np.array([1.0, 1.5, 1.9])
+    gdp_profile = pd.DataFrame({"years": [700, 710, 720], "values": [1.0, 1.5, 1.9]})
+    missing_years = [y for y in model_horizon if y not in set(gdp_profile.years)]
+    if missing_years:
+        gdp_profile = pd.concat(
+            [
+                gdp_profile,
+                pd.DataFrame(
+                    {"years": missing_years, "values": [np.nan] * len(missing_years)}
+                ),
+            ]
+        )
+        # Interpolate missing years
+        gdp_profile = (
+            gdp_profile.sort_values(by="years")
+            .set_index("years")
+            .interpolate(how="index", limit_direction="both")
+            .reset_index()
```

Comment:
It seems there is no need to "reset_index()" here, as in the following line "years" will be set as index again.

Reply:
It is, as this is required to get the dataframe in the same format if no missing years are added i.e. without the "if" statement.